account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/UpdateEmailHandler.java

```diff
@@ -0,0 +1,88 @@
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.oauth2.sdk.id.Subject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.di.accountmanagement.entity.UpdateEmailRequest;
+import uk.gov.di.accountmanagement.services.AwsSqsClient;
+import uk.gov.di.authentication.shared.entity.ErrorResponse;
+import uk.gov.di.authentication.shared.entity.NotificationType;
+import uk.gov.di.authentication.shared.entity.NotifyRequest;
+import uk.gov.di.authentication.shared.services.ConfigurationService;
+import uk.gov.di.authentication.shared.services.DynamoService;
+
+import java.util.Map;
+
+import static uk.gov.di.authentication.shared.helpers.ApiGatewayResponseHelper.generateApiGatewayProxyErrorResponse;
+import static uk.gov.di.authentication.shared.helpers.ApiGatewayResponseHelper.generateApiGatewayProxyResponse;
+
+public class UpdateEmailHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final DynamoService dynamoService;
+    private final AwsSqsClient sqsClient;
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpdateEmailHandler.class);
+
+    public UpdateEmailHandler() {
+        ConfigurationService configurationService = new ConfigurationService();
+        this.dynamoService = new DynamoService(configurationService);
+        this.sqsClient =
+                new AwsSqsClient(
+                        configurationService.getAwsRegion(),
+                        configurationService.getEmailQueueUri(),
+                        configurationService.getSqsEndpointUri());
+    }
+
+    public UpdateEmailHandler(DynamoService dynamoService, AwsSqsClient sqsClient) {
+        this.dynamoService = dynamoService;
+        this.sqsClient = sqsClient;
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+        LOGGER.info("UpdateEmailHandler received request");
+        LOGGER.info(
+                "Authorizer parameters received: {}", input.getRequestContext().getAuthorizer());
+        try {
+            UpdateEmailRequest updateInfoRequest =
+                    objectMapper.readValue(input.getBody(), UpdateEmailRequest.class);
+            Subject subjectFromEmail =
+                    dynamoService.getSubjectFromEmail(updateInfoRequest.getExistingEmailAddress());
+            Map<String, Object> authorizerParams = input.getRequestContext().getAuthorizer();
+
+            if (!authorizerParams.containsKey("principalId")) {
```

Comment:
there is a helper method for this that @cj1993 has added which is probably worth while using

Reply:
Going to leave it as it is for now